xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java

```diff
@@ -302,4 +293,25 @@ private String computeAltAttributeValue(ResourceReference reference)
         }
         return label;
     }
+
+    private boolean shouldIgnoreAll(TagStack stack)
+    {
+        Optional<Iterator<Object>> stackParameterIterator =
+            Optional.ofNullable(stack.getStackParameterIterator(NON_GENERATED_CONTENT_STACK));
+        boolean inNonGeneratedContent =
+            stackParameterIterator
+                .map(iterator -> {
+                    while (iterator.hasNext()) {
+                        NonGeneratedContentStackValue value = (NonGeneratedContentStackValue) iterator.next();
+                        // Stops at the first value which is associated with a macro.
```

Comment:
Why is a metadata element associated with a macro? Couldn't this also be independent of a macro?

Reply:
Indeed, I just noticed that this can also be used for macro parameters.